x/profile/client/cli/tx.go

```diff
@@ -29,118 +29,56 @@ func GetTxCmd(_ string, cdc *codec.Codec) *cobra.Command {
-			txBldr := auth.NewTxBuilderFromCLI(inBuf).WithTxEncoder(utils.GetTxEncoder(cdc))
-			cliCtx := context.NewCLIContextWithInput(inBuf).WithCodec(cdc)
-
-			picture := GetFlagValueOrNilOnDefault(flagProfilePic)
-			cover := GetFlagValueOrNilOnDefault(flagProfileCover)
-			var pictures *types.Pictures
-			if picture != nil && cover != nil {
-				pictures = types.NewPictures(picture, cover)
-			}
-			name := GetFlagValueOrNilOnDefault(flagName)
-			surname := GetFlagValueOrNilOnDefault(flagSurname)
-			bio := GetFlagValueOrNilOnDefault(flagBio)
-
-			msg := types.NewMsgCreateProfile(args[0], name, surname, bio, pictures, cliCtx.FromAddress)
-
-			return utils.GenerateOrBroadcastMsgs(cliCtx, txBldr, []sdk.Msg{msg})
-		},
-	}
-
-	cmd.Flags().String(flagName, "default", "Name of the profile")
-	cmd.Flags().String(flagSurname, "default", "Surname of the profile")
-	cmd.Flags().String(flagBio, "default", "Biography of the profile")
-	cmd.Flags().String(flagProfilePic, "default", "Profile related picture")
-	cmd.Flags().String(flagProfileCover, "default", "Profile related cover picture")
-
-	return cmd
-}
-
-// GetCmdEditProfile is the CLI command for editing an profile
-func GetCmdEditProfile(cdc *codec.Codec) *cobra.Command {
-	cmd := &cobra.Command{
-		Use:   "edit",
-		Short: "Edit an existent profile",
-		Long: fmt.Sprintf(`
-Edit an existing profile specifying the new moniker, name, surname, bio, a profile picture and cover.
-Every data is optional.
-
-All the attributes will not accept the string "default" as a value because it's reserved for internal operations.
-
-E.g (with all the other optional fields)
-%s tx profiles edit \
-    --moniker "DiCapLeo" \
-	--name "Leo" \
-	--surname "Di Cap" \
-	--bio "Hollywood actor. Proud environmentalist" \
-	--picture "https://profilePic.jpg"
-	--cover "https://profileCover.jpg"
 `, version.ClientName),
 		Args: cobra.ExactArgs(0),
 		RunE: func(cmd *cobra.Command, args []string) error {
 			inBuf := bufio.NewReader(cmd.InOrStdin())
 			txBldr := auth.NewTxBuilderFromCLI(inBuf).WithTxEncoder(utils.GetTxEncoder(cdc))
 			cliCtx := context.NewCLIContextWithInput(inBuf).WithCodec(cdc)
 
-			picture := GetFlagValueOrNilOnDefault(flagProfilePic)
-			cover := GetFlagValueOrNilOnDefault(flagProfileCover)
-			newMoniker := GetFlagValueOrNilOnDefault(flagNewMoniker)
-			name := GetFlagValueOrNilOnDefault(flagName)
-			surname := GetFlagValueOrNilOnDefault(flagSurname)
-			bio := GetFlagValueOrNilOnDefault(flagBio)
+			picture := viper.GetString(flagProfilePic)
+			cover := viper.GetString(flagProfileCover)
+			moniker := viper.GetString(flagMoniker)
+			name := viper.GetString(flagName)
+			surname := viper.GetString(flagSurname)
+			bio := viper.GetString(flagBio)
 
-			msg := types.NewMsgEditProfile(newMoniker, name, surname, bio, picture, cover, cliCtx.FromAddress)
+			msg := types.NewMsgSaveProfile(moniker, &name, &surname, &bio, &picture, &cover, cliCtx.FromAddress)
```

Comment:
Instead of simply creating a `MsgSaveProfile` with the provided values, I think it might be better to: 

1. Query the existing profile. 
2. If the profile has some values, use them if the new ones are set to `nil`. 
3. Save the edited profile. 

Point 1 can be achieved using 

```golang
route := fmt.Sprintf("custom/%s/%s/%s", types.QuerierRoute, types.QueryProfile, cliCtx.FromAddress.String())
res, _, err := cliCtx.QueryWithData(route, nil)
if err != nil {
  fmt.Printf("Could not find a profile with moniker %s \n", args[0])
  return nil
}

var out types.Profile
cdc.MustUnmarshalJSON(res, &out)
```

For point 2, I suggest creating a set of methods called `EditXXX` that replaces the exiting values of they are not `nil`. For example: 

```golang
profile := profile.EditMoniker(newMoniker).EditName(newName).EditSurname(newSurname)
```

where

```golang
func (profile Profile) EditName(newName *string) Profile {
  if newName != nil && len(newName) > 0 {
    profile.Name = newName
  }
  return profile
}
```

Reply:
I think we could do this inside the `handler`, I'll edit it and commit asap

Reply:
While I'm editing the handler I realized that, with a method like `editName` a user will never be able do "delete" a field, if he want to do so.

Reply:
@bragaz You're right. Good point!